gammapy/time/lombscargle_gammapy.py

```diff
@@ -0,0 +1,171 @@
+# from lombscargle
+from astropy.stats import LombScargle
+# from PyAstronomy.pyTiming.pyPeriod import Gls
+# from PyAstronomy.pyTiming import pyPeriod
+import numpy as np
+# import matplotlib
+import matplotlib.pyplot as plt
```

Comment:
matplotlib and scipy are optional dependencies of Gammapy.
Usually we put those imports as "delayed" imports into the functions where it's used.
Having them at the module level (i.e. at the top) means that you can't even import this module if the dependency is not installed.
One should be able to use your `lombscargle` function to compute something even if matplotlib isn't installed.
Is it clear? Can you change these imports?